### PR TITLE
Update to Go 1.18

### PR DESCRIPTION
**Pull request by webrouse**
Part of: https://keboola.atlassian.net/browse/KAC-132

Changes:
- Updated to Go 1.18
- Temporary disable linters that are incompatible.

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: GitHub Actions
 on: [ push ]
 env:
-  GO_VERSION: 1.17
+  GO_VERSION: 1.18
```
Comment (webrouse) [inline .github/workflows/push.yml:4]: Go `1.18` bolo vydane o `15.3`, a zatial nerobili ziadne opravy, iba nejake drobne security fixes.
https://go.dev/doc/devel/release

Nejaké generické typy sa mi hodia podla vsetkeho pri telemetrii a aj refactoringu.

File: build/ci/golangci.yml [hunk 2]
```diff
@@ -44,9 +44,12 @@ linters-settings:
 
 # https://golangci-lint.run/usage/linters
 linters:
+  disable-all: true
```
Comment (webrouse) [inline build/ci/golangci.yml:51]: Predtym tu bolo `enabled` aj `disabled` (niektore linters su enabled by default), pridal som tu `disable-all`, a vymenoval vsetky linters v `enabled` (a oznacil komentarom vypnute), ... takto je to jednoznacne.

File: build/ci/golangci.yml [hunk 3]
```diff
@@ -44,9 +44,12 @@ linters-settings:
...
+  # Some linters are not compatible with go 1.18, so they are temporary disabled.
+  # See https://github.com/golangci/golangci-lint/issues/2649
   enable:
     - asciicheck
-    - bodyclose
+    #- bodyclose # Not compatible with go 1.18, temporary disable
```
Comment (webrouse) [inline build/ci/golangci.yml:56]: Niektore `linters` nefunguju s `go 1.18`,
Tu je tabulka: https://github.com/golangci/golangci-lint/issues/2649

Comment (webrouse) [reply build/ci/golangci.yml:56]: image

File: build/ci/golangci.yml [hunk 4]
```diff
@@ -16,7 +16,11 @@ linters-settings:
       - github.com/keboola/keboola-as-code/*
   # Gci - improved version of goimports
   gci:
-    local-prefixes: github.com/keboola/keboola-as-code
+    sections:
+      - standard # Captures all standard packages if they do not match another section.
+      - default # Contains all imports that could not be matched to another section type.
+      - prefix(github.com/keboola/keboola-as-code)
+
```
Comment (webrouse) [inline build/ci/golangci.yml:23]: Zmenil sa syntax konfiguracie, je to to iste.

File: build/ci/golangci.yml (hunk 3 above)
Comment (webrouse) [reply build/ci/golangci.yml:56]: Cakam, ze behom par mesiacov budu fixnute aj dalsie a vratime to spat.

**Review by JakubMatejka — APPROVED** [review]
👍

**Pull request merged**